### ISSUE TITLE
channels_last/channels_last_3d memory format not supported for some modules on ROCm that should be supported on CUDA

### ISSUE THREAD

**Issue by mikaylagawarecki** [pytorch/pytorch#70125]
### 🐛 Describe the bug

 #69317 is adding a test `test_memory_format` to `test/test_modules.py` to determine whether outputs are valid and in the correct memory format when the inputs and modules are converted to various memory formats. However, for `dtype=float32` on the CI ROCm cuda test, the following modules do not produce the expected behavior:

- Batchnorm2d
- Batchnorm3d
- Conv2d
- Conv3d
- ConvTransponse2d

i.e. when either input.to(memory_format=torch.channels_last) (or torch.channels_last_3d where relevant) (or module.to(memory_format=torch.channels_last) when the module has state) the assert output.is_contiguous(memory_format=torch.channels_last) fails

logs here: https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-bionic-rocm4.3.1-py3.6-test1/14485/console

### Versions

ROCm CI

cc @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport @KyleCZH @jbschlosser

**Comment by hongxiayang** [comment]
This should be closed.
